src/libmongoc/src/mongoc/mongoc-client.c

```diff
@@ -3080,3 +3081,40 @@ mongoc_client_enable_auto_encryption (mongoc_client_t *client,
    }
    return _mongoc_cse_client_enable_auto_encryption (client, opts, error);
 }
+
+bool
+mongoc_client_set_timeout_ms (mongoc_client_t *client,
+			      int64_t timeout,
+			      bson_error_t *error) {
+   if (!client) {
+      bson_set_error (error,
+		      MONGOC_ERROR_CLIENT,
+		      MONGOC_ERROR_CLIENT_NULL,
+		      "NULL client parameter");
+	 return false;		      
+   }
+
+   if (timeout < 0) {
+      bson_set_error (error,
+		      MONGOC_ERROR_TIMEOUT,
+		      MONGOC_ERROR_TIMEOUT_INVALID,
+		      "timeoutMS must be a non-negative integer");
+	 return false;
+   }
+
+   client->timeout_ms = timeout;
+   client->timeout_set = true;
+
+   return true;
```

Comment:
There's a timeout struct in `mongoc-timeout.h` that ensures a negative timeout is never set and keeps `is_set` and `timeout_ms` in sync among other things. We should use that. Here and in the other setters/getters.

Reply:
Oh nice, I didn't realize we had that type.  Switched to use it.